src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/task/BaseTasklet.java

```diff
@@ -30,6 +30,7 @@
     @Value("${logging-component-name}")
     private String logComponentName;
 
+    @SuppressWarnings("java:S112")
```

Comment:
Suppressing Sonar issue _"Define and throw a dedicated exception instead of using a generic one"_

Reply:
May be we could create an LrdDataLoadException class and check which line exactly is throwing the exception and convert it into the new exception type and throw it. 
```
if(conditionFails) {
     throw new LrdDataLoadException("Exception Message");
}
```

Reply:
Hi @viswajit-kuna - the only exception thrown in the method is `FailedToCreateRouteException` - so I will update it to this exception

Reply:
Addressed here: https://github.com/hmcts/rd-location-ref-data-load/pull/176/commits/b3a35015febc30e42a194f5533df03823deab7e8